### ISSUE TITLE
Rename module_systems 'es6' to 'esm'

### ISSUE THREAD

**Issue by dmmulroy** [melange-re/melange#1066]
Currently `melange`, via `dune` can be configured to compile to common js (cjs) to EcmaScript Modules (esm). Right now the option incorrectly uses `es6` to output esm. ES6, also known as EcmaScript 2015, is a versioned specification of the EcmaScript language (https://262.ecma-international.org/6.0/).

We should update Dune and the Melange documentation to reflect this.

**Comment by anmonteiro** [comment]
yeah I think this is a good idea, but it'll involve some coordination with dune. I'll get the ball rolling.
